### PR TITLE
🩹Fix: Adaptor middleware duplicates cookies

### PR DESCRIPTION
**Pull request by sigmundxia**
# Description

In `adapter.go`, add a step to clear existing cookies before introducing new ones in the `HTTPMiddleware` section, so that cookies are not duplicated after passing through the middleware.

Fixes [#3089](https://github.com/gofiber/fiber/issues/3089)

## Type of change

- [x] Code consistency (non-breaking change which improves code reliability and robustness)

## Checklist

Before you submit your pull request, please make sure you meet these requirements:

- [x] Followed the inspiration of the Express.js framework for new functionalities, making them similar in usage.
- [x] Conducted a self-review of the code and provided comments for complex or critical parts.
- [x] Updated the documentation in the `/docs/` directory for [Fiber's documentation](https://docs.gofiber.io/).
- [x] Added or updated unit tests to validate the effectiveness of the changes or new features.
- [x] Ensured that new and existing unit tests pass locally with the changes.
- [x] Verified that any new dependencies are essential and have been agreed upon by the maintainers/community.
- [x] Aimed for optimal performance with minimal allocations in the new code.
- [x] Provided benchmarks for the new code to analyze and improve upon.

## Commit formatting

Please use emojis in commit messages for an easy way to identify the purpose or intention of a commit. Check out the emoji cheatsheet here: [CONTRIBUTING.md](https://github.com/gofiber/fiber/blob/master/.github/CONTRIBUTING.md#pull-requests-or-commits)

### CODE REVIEW

File: middleware/adaptor/adaptor_test.go [hunk 1]
```diff
@@ -200,6 +201,49 @@ func Test_HTTPMiddleware(t *testing.T) {
...
+		switch cookie.Name {
+		case "cookieOne":
+			require.Equal(t, "valueCookieOne", cookie.Value)
+		case "cookieTwo":
+			require.Equal(t, "valueCookieTwo", cookie.Value)
+		default:
+			t.Error("unexpected cookie key")
+		}
+	}
+}
```
[the remark this answers is not in the document]
Comment (gaby) [reply middleware/adaptor/adaptor_test.go:245]: @sigmundxia Thoughts on the extra cases?

[the remark this answers is not in the document]
Comment (sigmundxia) [reply middleware/adaptor/adaptor_test.go]: @gaby Okay, I think rabbit's suggestion is spot on and have changed it accordingly😊

Let me know if there is anything else I can do to improve it!

File: middleware/adaptor/adaptor_test.go [hunk 2]
```diff
@@ -200,6 +201,81 @@ func Test_HTTPMiddleware(t *testing.T) {
...
+	}
+
+	app := fiber.New()
+	app.Use(HTTPMiddleware(nethttpMW))
+	app.Post("/", func(c fiber.Ctx) error {
+		// RETURNING CURRENT COOKIES TO RESPONSE
+		var cookies []string = strings.Split(c.Get(cookieHeader), "; ")
+		for _, cookie := range cookies {
+			c.Set(setCookieHeader, cookie)
+		}
```
[the remark this answers is not in the document]
Comment (sigmundxia) [reply middleware/adaptor/adaptor_test.go:230]: `c.Set` do not return error, so I suggest to keep the same. Thank you rabbit!

**Review by ReneWerner87 — APPROVED** [review]

**Review by efectn — APPROVED** [review]

**Review by gaby — APPROVED** [review]
LGTM

**Pull request merged**